### PR TITLE
BUG: alma fixes for list of keywords, fixes #2094

### LINKED ISSUE #2094: ALMA: Cannot select multiple science keywords
I attempted this query:
```python
result = Alma().query(payload={'spatial_resolution':'<0.1', 'science_keyword':
                               ['High-mass star formation',
                                'Disks around high-mass stars']},
                      public=True)
```
resulting in an exception `DALQueryError: ADQL syntax error: Encountered " <S_IDENTIFIER> "High "" at line 1, column 81.`

The problem was a malformed ADQL query:
```
TAP query: select * from ivoa.obscore WHERE spatial_resolution<=0.1 AND science_keyword='['High-mass star formation', 'Disks around high-mass stars']' AND science_observation='T' AND data_rights='Public'
```

The solution is to do the query correctly, e.g.:
```python
AA.tap.search("select * from ivoa.obscore WHERE spatial_resolution<=0.1 AND science_keyword in ('High-mass star formation', 'Disks around high-mass stars') AND science_observation='T' AND data_rights='Public'")
```

### PR DESCRIPTION
**Pull request by at88mph**
ADQL not handling lists of strings properly.  Existing `OR` logic was used instead of introducing `IN` clause.

fixes #2094

### CODE REVIEW

**Comment by at88mph** [comment]
The failed `test_timer.py` in 3.8 runs fine locally.  Does this need to be kicked and rerun or is it a genuine failure?

File: astroquery/alma/tests/test_alma.py [hunk 1]
```diff
@@ -173,6 +173,15 @@ def test_gen_str_sql():
         "(proposal_id LIKE '2012.%' OR proposal_id LIKE '2013._3%')"
 
 
+def test_gen_array_sql():
+    common_select = "select * from ivoa.obscore WHERE "
```
Comment (bsipocz) [inline astroquery/alma/tests/test_alma.py:177]: extreme nitpick: maybe add a comment that this tests the regression from #2094?

**Review by bsipocz — APPROVED** [review]
Please add a changelog entry, and maybe remove the last commit that has the goal of re-triggering CI (that unrelated test is somewhat flaky, I'll try to cook up a solution so it will show up less often as a false failure in the future)

File: astroquery/alma/tests/test_alma.py (hunk 1 above)
Comment (at88mph) [reply astroquery/alma/tests/test_alma.py:177]: Sure, thanks @bsipocz .  Will the `CHANGES.rst` entry fall under the `0.4.7` heading?

Comment (bsipocz) [reply astroquery/alma/tests/test_alma.py:177]: yes, please

**Comment by bsipocz** [comment]
We generally avoid merge commits for the purpose of updating a branch, so I'll be removing that one prior merging the PR. Otherwise, it all looks good.

**Comment by at88mph** [comment]
Apologies.  Is that documented somewhere and I missed it?

File: CHANGES.rst [hunk 2]
```diff
@@ -20,6 +20,11 @@ hsa
 Service fixes and enhancements
 ------------------------------
 
+alma
+^^^^
+
+- Fixed a regression to handle arrays of string input for the ``query`` methods. [#2094]
```
Comment (bsipocz) [inline CHANGES.rst:26]: the number in the changelog should point to the number of the PR. I'm fixing this directly. (and this should be pulled into the dev docs highlights, too as it's also one deep in the weeds)

**Comment by bsipocz** [comment]
Tests have passed here already, I've only removed the last merge commit, so I'm going ahead and merge the PR now.

Thanks @at88mph!

**Pull request merged**